drivers/SmartThings/zigbee-water-leak-sensor/src/sinope/init.lua

```diff
@@ -0,0 +1,67 @@
+-- Copyright 2023 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local capabilities = require "st.capabilities"
+local zcl_clusters = require "st.zigbee.zcl.clusters"
+local IASZone = zcl_clusters.IASZone
+
+local SINOPE_TECHNOLOGIES_MFR_STRING = "Sinope Technologies"
+
+local generate_event_from_zone_status = function(driver, device, zone_status, zb_rx)
+  local event
+  if zone_status.value & 1 == 1 then
+    event = capabilities.waterSensor.water.wet()
+  else
+    event = capabilities.waterSensor.water.dry()
+  end
+
+  if event ~= nil then
+    device:emit_event(event)
+  end
+end
```

Comment:
so the distinction here is that you're only checking `zone_status:is_alarm1_set()` as opposed to the default driver, which checks for `alarm2` as well, because it looks like the device sends an event with the alarm2 field set when the sensor is disconnected?

Sounds good, but can you use the `is_alarm1_set()` function of the zone status object, for clarity's sake?

Reply:
Done